### ISSUE TITLE
Could not publish deno 1.11.3 on crates.io

### ISSUE THREAD

**Issue by ry** [denoland/deno#11180]
```
# cargo publish
    Updating crates.io index
   Packaging deno v1.11.3 (/Users/ry/src/deno/cli)
   Verifying deno v1.11.3 (/Users/ry/src/deno/cli)
  Downloaded twoway v0.2.2
  Downloaded env_logger v0.8.4
  Downloaded swc_ecma_visit v0.32.1
  Downloaded lsp-types v0.89.2
  Downloaded swc_common v0.10.21
  Downloaded indexmap v1.7.0
  Downloaded swc_ecma_parser v0.60.2
  Downloaded bumpalo v3.7.0
  Downloaded fd-lock v2.0.0
  Downloaded dprint-plugin-json v0.12.1
  Downloaded dprint-core v0.43.0
  Downloaded rustyline v8.2.0
  Downloaded swc_ecma_utils v0.37.2
  Downloaded swc_ecma_ast v0.47.0
  Downloaded swc_ecma_dep_graph v0.28.1
  Downloaded swc_ecmascript v0.43.0
  Downloaded deno_runtime v0.18.1
  Downloaded swc_ecma_visit v0.33.0
  Downloaded swc_ecma_parser v0.61.1
  Downloaded dprint-plugin-typescript v0.46.2
  Downloaded dprint-swc-ecma-ast-view v0.21.1
  Downloaded 21 crates (2.0 MB) in 1.24s
   Compiling libc v0.2.97
   Compiling proc-macro2 v1.0.27
   Compiling unicode-xid v0.2.2
   Compiling syn v1.0.65
   Compiling cfg-if v1.0.0
   Compiling autocfg v1.0.1
   Compiling serde_derive v1.0.126
   Compiling serde v1.0.126
   Compiling version_check v0.9.3
   Compiling once_cell v1.8.0
   Compiling memchr v2.4.0
   Compiling log v0.4.14
   Compiling bitflags v1.2.1
   Compiling lazy_static v1.4.0
   Compiling itoa v0.4.7
   Compiling matches v0.1.8
   Compiling tinyvec_macros v0.1.0
   Compiling ryu v1.0.5
   Compiling percent-encoding v2.1.0
   Compiling smallvec v1.6.1
   Compiling serde_json v1.0.64
   Compiling either v1.6.1
   Compiling fnv v1.0.7
   Compiling ppv-lite86 v0.2.10
   Compiling proc-macro-hack v0.5.19
   Compiling byteorder v1.4.3
   Compiling unicode-width v0.1.8
   Compiling getrandom v0.1.16
   Compiling pin-project-lite v0.2.7
   Compiling regex-syntax v0.6.25
   Compiling futures-core v0.3.15
   Compiling siphasher v0.3.5
   Compiling futures-task v0.3.15
   Compiling proc-macro-nested v0.1.7
   Compiling futures-channel v0.3.15
   Compiling futures-sink v0.3.15
   Compiling scopeguard v1.1.0
   Compiling pin-utils v0.1.0
   Compiling slab v0.4.3
   Compiling futures-io v0.3.15
   Compiling bytes v1.0.1
   Compiling ident_case v1.0.1
   Compiling strsim v0.9.3
   Compiling new_debug_unreachable v1.0.4
   Compiling precomputed-hash v0.1.1
   Compiling base64 v0.13.0
   Compiling if_chain v1.0.1
   Compiling base64 v0.11.0
   Compiling stable_deref_trait v1.2.0
   Compiling cfg-if v0.1.10
   Compiling scoped-tls v1.0.0
   Compiling anyhow v1.0.41
   Compiling typenum v1.13.0
   Compiling lexical-core v0.7.6
   Compiling static_assertions v1.1.0
   Compiling untrusted v0.7.1
   Compiling opaque-debug v0.3.0
   Compiling httparse v1.4.1
   Compiling termcolor v1.1.2
   Compiling core-foundation-sys v0.8.2
   Compiling bit-vec v0.6.3
   Compiling fixedbitset v0.2.0
   Compiling encoding_rs v0.8.28
   Compiling foreign-types-shared v0.1.1
   Compiling alloc-no-stdlib v2.0.1
   Compiling crc32fast v1.2.1
   Compiling async-trait v0.1.50
   Compiling tower-service v0.3.1
   Compiling try-lock v0.2.3
   Compiling block v0.1.6
   Compiling pkg-config v0.3.19
   Compiling adler v1.0.2
   Compiling ipnet v2.3.1
   Compiling ahash v0.4.7
   Compiling httpdate v1.0.1
   Compiling cfg_aliases v0.1.1
   Compiling utf-8 v0.7.6
   Compiling range-alloc v0.1.2
   Compiling copyless v0.1.5
   Compiling unicode-segmentation v1.7.1
   Compiling profiling v1.0.3
   Compiling bumpalo v3.7.0
   Compiling fallible-streaming-iterator v0.1.9
   Compiling fallible-iterator v0.2.0
   Compiling proc-macro2 v0.4.30
   Compiling mime v0.3.16
   Compiling retain_mut v0.1.3
   Compiling unicode-xid v0.1.0
   Compiling crossbeam-utils v0.8.5
   Compiling syn v0.15.44
   Compiling match_cfg v0.1.0
   Compiling build_const v0.2.2
   Compiling pulldown-cmark v0.8.0
   Compiling same-file v1.0.6
   Compiling data-encoding v2.3.2
   Compiling quick-error v1.2.3
   Compiling linked-hash-map v0.5.4
   Compiling swc_bundler v0.40.0
   Compiling endian-type v0.1.2
   Compiling ucd-trie v0.1.3
   Compiling unchecked-index v0.2.2
   Compiling convert_case v0.4.0
   Compiling ansi_term v0.11.0
   Compiling utf8parse v0.2.0
   Compiling remove_dir_all v0.5.3
   Compiling humantime v2.1.0
   Compiling vec_map v0.8.2
   Compiling strsim v0.8.0
   Compiling relative-path v1.4.0
   Compiling radix_fmt v1.0.0
   Compiling text-size v1.1.0
   Compiling dissimilar v1.0.2
   Compiling shell-escape v0.1.5
   Compiling instant v0.1.9
   Compiling thread_local v1.1.3
   Compiling indexmap v1.7.0
   Compiling num-traits v0.2.14
   Compiling futures-macro v0.3.15
   Compiling futures-util v0.3.15
   Compiling num-integer v0.1.44
   Compiling num-bigint v0.2.6
   Compiling tokio v1.7.1
   Compiling miniz_oxide v0.4.4
   Compiling tracing-core v0.1.18
   Compiling unicode-bidi v0.3.5
   Compiling data-url v0.1.0
   Compiling ahash v0.7.4
   Compiling generic-array v0.14.4
   Compiling proc-macro-error-attr v1.0.4
   Compiling unicase v2.6.0
   Compiling proc-macro-error v1.0.4
   Compiling tinyvec v1.2.0
   Compiling form_urlencoded v1.0.1
   Compiling nibble_vec v0.1.0
   Compiling textwrap v0.11.0
   Compiling fxhash v0.2.1
   Compiling phf_shared v0.8.0
   Compiling lock_api v0.4.4
   Compiling owning_ref v0.4.1
   Compiling http v0.2.4
   Compiling input_buffer v0.4.0
   Compiling codespan-reporting v0.11.1
   Compiling bit-set v0.5.2
   Compiling foreign-types v0.3.2
   Compiling alloc-stdlib v0.2.1
   Compiling hashbrown v0.9.1
   Compiling wgpu-core v0.8.1
   Compiling heck v0.3.3
   Compiling walkdir v2.3.2
   Compiling crc v1.8.1
   Compiling lru-cache v0.1.2
   Compiling pest v2.1.3
   Compiling tracing v0.1.26
   Compiling radix_trie v0.2.1
   Compiling storage-map v0.3.0
   Compiling brotli-decompressor v2.3.1
   Compiling unicode-normalization v0.1.19
   Compiling quote v1.0.9
   Compiling http-body v0.4.2
   Compiling getrandom v0.2.3
   Compiling jobserver v0.1.22
   Compiling num_cpus v1.13.0
   Compiling parking_lot_core v0.8.3
   Compiling signal-hook-registry v1.4.0
   Compiling fslock v0.1.6
   Compiling which v4.1.0
   Compiling cpufeatures v0.1.5
   Compiling malloc_buf v0.0.6
   Compiling raw-window-handle v0.3.3
   Compiling socket2 v0.4.0
   Compiling hostname v0.3.1
   Compiling atty v0.2.14
   Compiling filetime v0.2.14
   Compiling fsevent-sys v4.0.0
   Compiling fd-lock v2.0.0
   Compiling semver-parser v0.10.2
   Compiling gpu-alloc-types v0.2.0
   Compiling gpu-descriptor-types v0.1.1
   Compiling nix v0.20.0
   Compiling aho-corasick v0.7.18
   Compiling twoway v0.2.2
   Compiling core-foundation v0.9.1
   Compiling crossbeam-channel v0.5.1
   Compiling quote v0.6.13
   Compiling flate2 v1.0.20
   Compiling brotli v3.3.0
   Compiling idna v0.2.3
   Compiling rand_core v0.6.3
   Compiling spirv_headers v1.5.0
   Compiling rand_core v0.5.1
   Compiling cc v1.0.68
   Compiling dashmap v4.0.2
   Compiling parking_lot v0.11.1
   Compiling rusty_v8 v0.22.3
   Compiling resolv-conf v0.7.0
   Compiling clap v2.33.3
   Compiling gpu-descriptor v0.1.1
   Compiling gpu-alloc v0.4.7
   Compiling core-graphics-types v0.1.1
   Compiling regex v1.4.3
   Compiling notify v5.0.0-pre.10
   Compiling block-buffer v0.9.0
   Compiling digest v0.9.0
   Compiling hashbrown v0.11.2
   Compiling rand_chacha v0.3.1
   Compiling rand_pcg v0.2.1
   Compiling rand_chacha v0.2.2
   Compiling ring v0.16.20
   Compiling objc_exception v0.1.2
   Compiling libsqlite3-sys v0.22.2
   Compiling sys-info v0.9.0
   Compiling sha-1 v0.9.6
   Compiling hashlink v0.7.0
   Compiling Inflector v0.11.4
   Compiling fancy-regex v0.5.0
   Compiling rand v0.8.4
   Compiling pmutil v0.5.3
   Compiling darling_core v0.10.2
   Compiling rand v0.7.3
   Compiling dlopen_derive v0.1.4
   Compiling swc_macros_common v0.3.3
   Compiling tempfile v3.2.0
   Compiling phf_generator v0.8.0
   Compiling dlopen v0.1.8
   Compiling objc v0.2.7
   Compiling tokio-macros v1.2.0
   Compiling swc_eq_ignore_macros v0.1.0
   Compiling is-macro v0.1.9
   Compiling pin-project-internal v1.0.7
   Compiling thiserror-impl v1.0.25
   Compiling enum-as-inner v0.3.3
   Compiling serde_repr v0.1.7
   Compiling auto_impl v0.4.1
   Compiling lspower-macros v0.2.1
   Compiling derive_more v0.99.14
   Compiling rustyline-derive v0.4.0
   Compiling string_cache_codegen v0.5.1
   Compiling phf_macros v0.8.0
   Compiling swc_visit_macros v0.2.3
   Compiling from_variant v0.1.3
   Compiling string_enum v0.3.1
   Compiling enum_kind v0.2.1
   Compiling swc_ecma_codegen_macros v0.5.2
   Compiling cocoa-foundation v0.1.0
   Compiling darling_macro v0.10.2
   Compiling thiserror v1.0.25
   Compiling pin-project v1.0.7
   Compiling swc_atoms v0.2.6
   Compiling phf v0.8.0
   Compiling darling v0.10.2
   Compiling swc_visit v0.2.4
   Compiling ast_node v0.7.3
   Compiling webpki v0.21.4
   Compiling sct v0.6.1
   Compiling rusqlite v0.25.3
   Compiling webpki-roots v0.21.1
   Compiling futures-executor v0.3.15
   Compiling futures v0.3.15
   Compiling url v2.2.2
   Compiling arrayvec v0.5.2
   Compiling string_cache v0.8.1
   Compiling uuid v0.8.2
   Compiling ron v0.6.4
   Compiling serde_urlencoded v0.7.0
   Compiling wgpu-types v0.8.0
   Compiling dprint-core v0.43.0
   Compiling dprint-core v0.42.0
   Compiling mio v0.7.13
   Compiling rustls v0.19.1
   Compiling want v0.3.0
   Compiling metal v0.22.0
   Compiling env_logger v0.8.4
   Compiling rustyline v8.2.0
   Compiling petgraph v0.5.1
   Compiling dprint-plugin-markdown v0.8.0
   Compiling sourcemap v6.0.1
   Compiling jsonc-parser v0.17.0
   Compiling lsp-types v0.89.2
   Compiling tungstenite v0.13.0
   Compiling naga v0.4.2
   Compiling lexical v5.2.2
   Compiling swc_common v0.10.21
   Compiling dprint-plugin-json v0.12.1
   Compiling tokio-util v0.6.7
   Compiling tokio-rustls v0.22.0
   Compiling async-compression v0.3.8
   Compiling tokio-stream v0.1.6
   Compiling trust-dns-proto v0.20.3
   Compiling tokio-tungstenite v0.14.0
   Compiling h2 v0.3.3
   Compiling swc_ecma_ast v0.46.0
   Compiling swc_ecma_ast v0.47.0
   Compiling gfx-hal v0.8.0
   Compiling trust-dns-resolver v0.20.3
   Compiling hyper v0.14.9
   Compiling gfx-backend-metal v0.8.2
   Compiling gfx-backend-empty v0.8.0
   Compiling lspower v1.1.0
   Compiling hyper-rustls v0.22.1
   Compiling reqwest v0.11.4
   Compiling swc_ecma_visit v0.32.1
   Compiling swc_ecma_visit v0.33.0
   Compiling swc_ecma_parser v0.60.2
   Compiling swc_ecma_utils v0.37.2
   Compiling swc_ecma_dep_graph v0.28.1
   Compiling swc_ecma_parser v0.61.1
   Compiling swc_ecma_transforms_base v0.18.0
   Compiling swc_ecma_codegen v0.58.0
   Compiling swc_ecmascript v0.43.0
   Compiling dprint-swc-ecma-ast-view v0.21.1
   Compiling swc_ecma_transforms_classes v0.4.0
   Compiling swc_ecma_transforms_react v0.21.0
   Compiling swc_ecma_transforms_typescript v0.22.0
   Compiling swc_ecma_transforms_optimization v0.23.0
   Compiling swc_ecma_transforms_proposal v0.20.0
   Compiling swc_ecma_transforms v0.53.0
   Compiling swc_ecmascript v0.39.0
   Compiling deno_doc v0.6.0
   Compiling deno_lint v0.7.0
error[E0277]: a value of type `Vec<dprint_swc_ecma_ast_view::TokenAndSpan>` cannot be built from an iterator over elements of type `swc_ecmascript::swc_ecma_parser::token::TokenAndSpan`
   --> /Users/ry/.cargo/registry/src/github.com-1ecc6299db9ec823/deno_lint-0.7.0/src/ast_parser.rs:187:63
    |
187 |     let tokens: Vec<TokenAndSpan> = lexer.clone().into_iter().collect();
    |                                                               ^^^^^^^ value of type `Vec<dprint_swc_ecma_ast_view::TokenAndSpan>` cannot be built from `std::iter::Iterator<Item=swc_ecmascript::swc_ecma_parser::token::TokenAndSpan>`
    |
    = help: the trait `FromIterator<swc_ecmascript::swc_ecma_parser::token::TokenAndSpan>` is not implemented for `Vec<dprint_swc_ecma_ast_view::TokenAndSpan>`

error[E0308]: mismatched types
   --> /Users/ry/.cargo/registry/src/github.com-1ecc6299db9ec823/deno_lint-0.7.0/src/linter.rs:163:57
    |
163 |       self.lint_program(file_name, &program, &comments, &tokens, &source_file);
    |                                                         ^^^^^^^ expected slice, found struct `Vec`
    |
    = note: expected reference `&[swc_ecmascript::swc_ecma_parser::token::TokenAndSpan]`
               found reference `&Vec<dprint_swc_ecma_ast_view::TokenAndSpan>`

error[E0308]: mismatched types
   --> /Users/ry/.cargo/registry/src/github.com-1ecc6299db9ec823/deno_lint-0.7.0/src/linter.rs:302:7
    |
302 |       program,
    |       ^^^^^^^ expected enum `dprint_swc_ecma_ast_view::swc_ecma_ast::Program`, found enum `swc_ecmascript::swc_ecma_ast::Program`
    |
    = note: expected reference `&dprint_swc_ecma_ast_view::swc_ecma_ast::Program`
               found reference `&swc_ecmascript::swc_ecma_ast::Program`
    = note: perhaps two different versions of crate `swc_ecma_ast` are being used?

error[E0308]: mismatched types
   --> /Users/ry/.cargo/registry/src/github.com-1ecc6299db9ec823/deno_lint-0.7.0/src/linter.rs:304:20
    |
304 |       tokens: Some(tokens),
    |                    ^^^^^^ expected struct `dprint_swc_ecma_ast_view::TokenAndSpan`, found struct `swc_ecmascript::swc_ecma_parser::token::TokenAndSpan`
    |
    = note: expected reference `&[dprint_swc_ecma_ast_view::TokenAndSpan]`
               found reference `&[swc_ecmascript::swc_ecma_parser::token::TokenAndSpan]`
    = note: perhaps two different versions of crate `swc_ecma_parser` are being used?

error[E0308]: mismatched types
   --> /Users/ry/.cargo/registry/src/github.com-1ecc6299db9ec823/deno_lint-0.7.0/src/rules.rs:121:39
    |
121 |       Module(m) => ProgramRef::Module(m.inner),
    |                                       ^^^^^^^ expected struct `swc_ecmascript::swc_ecma_ast::Module`, found struct `dprint_swc_ecma_ast_view::swc_ecma_ast::Module`
    |
    = note: expected reference `&swc_ecmascript::swc_ecma_ast::Module`
               found reference `&dprint_swc_ecma_ast_view::swc_ecma_ast::Module`
    = note: perhaps two different versions of crate `swc_ecma_ast` are being used?

error[E0308]: mismatched types
   --> /Users/ry/.cargo/registry/src/github.com-1ecc6299db9ec823/deno_lint-0.7.0/src/rules.rs:122:39
    |
122 |       Script(s) => ProgramRef::Script(s.inner),
    |                                       ^^^^^^^ expected struct `swc_ecmascript::swc_ecma_ast::Script`, found struct `dprint_swc_ecma_ast_view::swc_ecma_ast::Script`
    |
    = note: expected reference `&swc_ecmascript::swc_ecma_ast::Script`
               found reference `&dprint_swc_ecma_ast_view::swc_ecma_ast::Script`
    = note: perhaps two different versions of crate `swc_ecma_ast` are being used?

error[E0599]: no method named `string_repr` found for enum `dprint_swc_ecma_ast_view::swc_ecma_ast::PropName` in the current scope
   --> /Users/ry/.cargo/registry/src/github.com-1ecc6299db9ec823/deno_lint-0.7.0/src/rules/adjacent_overload_signatures.rs:239:19
    |
239 |         inner.key.string_repr().map(|k| {
    |                   ^^^^^^^^^^^ method not found in `dprint_swc_ecma_ast_view::swc_ecma_ast::PropName`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
note: `StringRepr` defines an item `string_repr`, perhaps you need to implement it
   --> /Users/ry/.cargo/registry/src/github.com-1ecc6299db9ec823/deno_lint-0.7.0/src/swc_util.rs:34:1
    |
34  | pub(crate) trait StringRepr {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 7 previous errors

Some errors have detailed explanations: E0277, E0308, E0599.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `deno_lint`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: failed to verify package tarball

Caused by:
  build failed
```

**Comment by piscisaureus** [comment]
Very strange. Tried `cargo publish --locked` but no luck with that either.

**Comment by dsherret** [comment]
I'm surprised it pulled in dprint-swc-ecma-ast-view v0.21.1 and swc_ecmascript v0.43.0. I thought it would have stuck to 0.21.0 and swc_ecmascript 0.39.0 due to the lock file. I upgraded the swc dependency in there yesterday and didn't notice any api changes so I just bumped the patch of dprint-swc-ecma-ast-view... looks like I should have published a minor release.

**Comment by lucacasonato** [comment]
@dsherret I guess the easiest fix is to update swc everywhere now? So deno_lint, deno_doc, and cli. Once that is done I can cut a new 1.11.4 release (mainly for #11181 though).

**Comment by dwalker109** [comment]
This is probably related, but a `deno upgrade` I did today (from a 1.4 version) failed with:

```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `"deno 1.11.3+f4bed10"`,
 right: `"deno 1.11.3"`', cli/upgrade.rs:272:3
```

Installing via Github gave me a canary version:

```
➜  bin deno --version
deno 1.11.3+f4bed10 (canary, x86_64-unknown-linux-gnu)
v8 9.1.269.35
typescript 4.3.2
```

**Comment by ry** [comment]
@dwalker109 That's a separate issue that we've fixed already -> https://github.com/denoland/deno/issues/11179

**Comment by lucacasonato** [comment]
1.11.4 was published to crates.io without issue
